pkg/credentialprovider/azure/azure_credentials_test.go

```diff
@@ -93,4 +93,17 @@ func Test(t *testing.T) {
 			t.Errorf("Missing expected registry: %s", registryName)
 		}
 	}
+
+	invalidImage := "invalidImage"
```

Comment:
pls add a separate unit test, you could refer to follow unit test as example: https://github.com/kubernetes/kubernetes/blob/cf22c56c7fbeaa166e5e86ebc4c0b75250b9843a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_standard_test.go#L52

Reply:
Added in a new function.

Reply:
@norshtein I mean also copy the test code style like following, it's more maintainable:
```
	tests := []struct {
		ID        string
		expected  string
		expectErr bool
	}{
		{
			ID:        "",
			expected:  "",
			expectErr: true,
		},
		{
			ID:        "foo/",
			expected:  "",
			expectErr: true,
		},
		{
			ID:        "foo/bar",
			expected:  "bar",
			expectErr: false,
		},
		{
			ID:        "foo/bar/baz",
			expected:  "baz",
			expectErr: false,
		},
	}

	for _, test := range tests {
		s, e := getLastSegment(test.ID)
		if test.expectErr && e == nil {
			t.Errorf("Expected err, but it was nil")
			continue
		}
		if !test.expectErr && e != nil {
			t.Errorf("Unexpected error: %v", e)
			continue
		}
		if s != test.expected {
			t.Errorf("expected: %s, got %s", test.expected, s)
		}
	}
}
```